core/templates/dev/head/pages/admin-page/services/admin-data.service.spec.ts

```diff
@@ -0,0 +1,59 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Tests for AdminDataService.
+ */
+
+require('pages/admin-page/services/admin-data.service.ts');
+
+describe('Admin Data Service', function() {
+  var AdminDataService = null;
+  var $httpBackend = null;
+  var sampleAdminData = {
+    property: 'value'
+  };
+
+  beforeEach(angular.mock.module('oppia'));
+
+  beforeEach(angular.mock.inject(function($injector) {
+    $httpBackend = $injector.get('$httpBackend');
+    AdminDataService = $injector.get('AdminDataService');
+  }));
+
+  it('should return the correct admin data', function() {
+    $httpBackend.expect('GET', '/adminhandler').respond(
+      200, sampleAdminData);
+
+    AdminDataService.getDataAsync().then(function(response) {
+      expect(response).toEqual(sampleAdminData);
```

Comment:
Re [this comment](https://github.com/oppia/oppia/pull/7186/files#r307195577):

Are we expecting `response` to be `sampleAdminData.data` here? (as this is the first call)

Reply:
The getDataAsync method already does the `response.data` stuff so just using response is fine here.

Reply:
@Jamesjay4199, calling `/adminhandler` returns `sampleAdminData` as a response, right? now in the `then(...` section, we are making the response to sampleAdminData.data (as you have explained [here](https://github.com/oppia/oppia/pull/7186/files#r306454927)), right?

Reply:
I don't think I'm understanding you here @DubeySandeep. 
Normally when an ajax request is made with $http, the data can be gotten from the response as `response.data`.
So in this case, the getDataAsync method returns a promise that resolves with `response.data`, so any call you make to the function will not need you to do the `.data` again.

In this test, we're mocking the http and expecting it to respond with sampleAdminData. There is no need for `sampleAdminData.data`. The `.data` is not defined in the sampleAdminData object.